pkg/tfsandbox/apply.go

```diff
@@ -10,24 +10,24 @@ import (
 
 // Apply runs the terraform apply command and returns the final state
 func (t *Tofu) Apply(ctx context.Context, logger Logger) (*State, error) {
-	state, err := t.apply(ctx, logger)
-	if err != nil {
-		return nil, err
-	}
+	state, applyErr := t.apply(ctx, logger)
 	s, err := newState(state)
 	if err != nil {
 		return nil, err
 	}
-	return s, nil
+	return s, applyErr
 }
 
 // Apply runs the terraform apply command and returns the final state
 func (t *Tofu) apply(ctx context.Context, logger Logger) (*tfjson.State, error) {
 	logWriter := newJSONLogPipe(ctx, logger)
 	defer logWriter.Close()
 
-	if err := t.tf.ApplyJSON(ctx, logWriter); err != nil {
-		return nil, fmt.Errorf("error running tofu apply: %w", err)
+	applyErr := t.tf.ApplyJSON(ctx, logWriter)
+	// if the apply failed just log it to debug logs and continue
```

Comment:
Style nit: may be good to lift to `func (t *Tofu) Apply(ctx context.Context, logger Logger) (*State, error) {` docstring as that's part of the contract.. This can now return non-nil errors together with non-nil State. That's a tad weird so worth documenting at the boundary.